lib/api/core/models/repositories/roleRepository.js

```diff
@@ -185,6 +185,11 @@ class RoleRepository extends Repository {
   validateAndSaveRole (role, options) {
     return role.validateDefinition()
       .then(() => {
+        if (role._id === 'anonymous' && !role.canLogIn()) {
+          throw new BadRequestError('Cannot remove login permission from anonymous role');
+        }
+      })
```

Comment:
I disagree, for the same reason as for profiles. The `Role.canLogIn` call excepts the `role` object to be well formed, which is checked by the `validateDefinition` call.